neuralprophet/forecaster.py

```diff
@@ -1107,7 +1112,7 @@ def fit(
         self.fitted = True
         return metrics_df
 
-    def predict(self, df: pd.DataFrame, decompose: bool = True, raw: bool = False):
+    def predict(self, df: pd.DataFrame, decompose: bool = True, raw: bool = False, auto_extend=False):
```

Comment:
To avoid the linked issues being raised, `auto_extend` should default to `True`, unless my logic is off?

Reply:
auto_extend for me means that we don't do the cutting at the end. So we cut all the time excpet for the case where it will cause teh whole dataframe to be nan. Can still reverse it